File: server/internal/model/plan_day.go
package model

import (
	"time"

	"github.com/google/uuid"
)

type PlanDay struct {
	ID        uuid.UUID `gorm:"type:uuid;default:gen_random_uuid();primaryKey"`
	PlanID    uuid.UUID `gorm:"type:uuid;index;not null"`
	Plan      Plan      `gorm:"foreignKey:PlanID"`
	Name      string    `gorm:"size:50;not null"`
	SortOrder int       `gorm:"default:0"`
	CreatedAt time.Time
	UpdatedAt time.Time
	Exercises []PlanExercise `gorm:"foreignKey:PlanDayID;constraint:OnDelete:CASCADE"`
}

File: mobile/src/models/PlanDay.ts
import { Model } from "@nozbe/watermelondb";
import type { Query, Relation } from "@nozbe/watermelondb";
import type { Associations } from "@nozbe/watermelondb/Model";
import { children, field, immutableRelation } from "@nozbe/watermelondb/decorators";

import type { Plan } from "./Plan";
import type { PlanExercise } from "./PlanExercise";
import type { Workout } from "./Workout";

export class PlanDay extends Model {
  static table = "plan_days";

  static associations: Associations = {
    plans: { type: "belongs_to", key: "plan_id" },
    plan_exercises: { type: "has_many", foreignKey: "day_id" },
    workouts: { type: "has_many", foreignKey: "plan_day_id" },
  };

  @field("plan_id") planId!: string;
  @field("name") name!: string;
  @field("sort_order") sortOrder!: number;
  @field("updated_at") updatedAt!: number;

  @immutableRelation("plans", "plan_id") plan!: Relation<Plan>;
  @children("plan_exercises") exercises!: Query<PlanExercise>;
  @children("workouts") workouts!: Query<Workout>;
}
